apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx

```diff
@@ -29,29 +33,51 @@ export const AddPatientExtended = () => {
     );
 
     const defaultValues = initial();
-
     const form = useForm<ExtendedNewPatientEntry>({
         defaultValues,
         mode: 'onBlur'
     });
+    const formIsDirty = form.formState.isDirty;
+
+    // setup navigation blocking for back button
+    const shouldBlockNavigation = formIsDirty && !bypassBlocker && !bypassModal;
+    // Fires when user attempts to navigate away from the page, i.e. via router link or back button
+    const onBlock = useCallback(() => {
+        // open modal to confirm cancel
+        setCancelModal(true);
+    }, []);
+    const blocker = useNavigationBlock({ shouldBlock: shouldBlockNavigation, onBlock });
 
-    const handleSave = form.handleSubmit(interaction.create);
+    const handleSaveClick = form.handleSubmit(interaction.create);
 
-    const handleCancel = () => {
-        if (value) {
-            handleCancelConfirm();
+    const handleCancelClick = useCallback(() => {
+        // if user clicked cancel button, we don't want to block navigation
+        setBypassBlocker(true);
+        if (bypassModal) {
+            handleModalConfirm();
         } else {
             setCancelModal(true);
         }
-    };
+    }, [bypassBlocker, bypassModal]);
 
-    const handleCancelConfirm = () => {
+    const handleModalConfirm = useCallback(() => {
+        blocker.proceed();
         navigate('/add-patient');
-    };
+    }, [blocker, navigate]);
 
-    const closeCancel = () => {
+    const handleModalClose = useCallback(() => {
+        blocker.reset();
         setCancelModal(false);
-    };
+        setBypassBlocker(false);
+    }, [blocker]);
+
+    // Reset the blocker after a successful submission
+    useEffect(() => {
+        if (interaction.status === 'created') {
+            blocker.reset();
+            setBypassBlocker(false);
```

Comment:
question (non-blocking):  These two lines are always called together, which tells me one of them might not be needed.  I don't see a clear path as to what could be consolidated but I wonder if the visibility of the cancel modal could be determined by `blocker.blocked` alone.

Reply:
You are right. I did simplify this somewhat and I see how these can be combined.